test/unit/abilities/forums_test.rb

```diff
@@ -1,312 +1,239 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 module Abilities
   class ForumsTest < ActiveSupport::TestCase
     def setup
       @provider = FactoryBot.create(:provider_account)
+      @admin_user = @provider.admins.first
+      # admin = FactoryBot.create(:admin, account: @provider)
+      @provider_user = FactoryBot.create(:user, account: @provider)
+
       @forum = @provider.forum
+      @topic = FactoryBot.create(:topic, forum: @forum)
+
+      @buyer = FactoryBot.create(:buyer_account, provider_account: @provider)
+      @buyer_user = FactoryBot.create(:user, account: @buyer)
```

Comment:
This seems to be bloating setup for all tests making them slower, maybe not a big deal. Not blocker definitely.

Reply:
OK I refactored it a bit more and also increased coverage by creating 2 branches: public and private forum.